src/pulsar_producer.erl

```diff
@@ -853,7 +853,7 @@ do_send_to_pulsar(State0) ->
      , replayq := Q
      , opts := ProducerOpts
      } = State0,
-    NumToPop = min(BatchSize, MaxInflight - InflightCalls0),
+    NumToPop = min(BatchSize, MaxInflight - map_size(Requests0)),
```

Comment:
should not change here.

Reply:
this is the number of queue items (calls) to pop from replayq

Reply:
it does need to change, to accommodate for the new way of viewing max inflight

otherwise, we might try to pop 0 or less items....

Reply:
... which was what I found out while testing locally.